src/NHibernate/Type/BooleanType.cs

```diff
@@ -13,6 +14,9 @@ namespace NHibernate.Type
 	[Serializable]
 	public class BooleanType : PrimitiveType, IDiscriminatorType
 	{
+		internal static readonly object TrueObject = true; // skipcq: CS-P1003
+		internal static readonly object FalseObject = false; // skipcq: CS-P1003
```

Comment:
Why is this internal?

Reply:
Yeah `protected` should suffice

Reply:
Well...

- `public` is not appropriate, I suppose, it should not be exposed to outside of the library.
- `private` restricts a usage to `BooleanType` class only, but the values are needed in an child class right now (maybe in more child classes in the future).
- `protected` - maybe, but why should we restrict the usage of these pre-created values to child classes only?
- everything above leads us to `internal`, because NHibernate uses a lot of booleans, and other parts of library might benefit from using of the values (I have not checked the whole codebase, obviously), and it is "safe", because not exposed for user code

If you strongly disagree, I may change access modifiers to 'protected', as it is minimum requirements for the code changes.

Reply:
changed to `protected`